pdf_build_src/process_markdowns.py

```diff
@@ -1,191 +1,418 @@
+            if file.endswith(".md"):
+                with open(os.path.join(root, file), 'r') as markdown:
+                    data = markdown.readlines()
+
+                for ind, line in enumerate(data):
+                    match = primary_pattern.search(line)
+
+                    if match:
+                        line = re.sub(primary_pattern,
+                                      match.group().split('](')[0][1:], line)
+
+                    data[ind] = line
+
+                with open(os.path.join(root, file), 'w') as markdown:
+                    markdown.writelines(data)
+
+
+def modify_changelog():
+    """Change first line of the changelog to markdown Heading 1.
+    This modification makes sure that in the pdf build, changelog is a new
+    chapter.
+    """
+    with open('src_copy/src/CHANGES.md', 'r') as file:
+        data = file.readlines()
+
+    data[0] = "# Changelog"
+
+    with open('src_copy/src/CHANGES.md', 'w') as file:
+        file.writelines(data)
+
+
+def correct_table(table, offset = [0.0, 0.0], debug=False):
+    """Create the corrected table.
+
+    Compute the number of characters maximal in each table column and reformat each 
+    row in the table to make sure the first and second rows of the table have enough
+    dashes (in proportion) and that fences are correctly aligned
+    for correct rendering in the generated PDF.
+
+    Parameters
+    ----------
+    table : list of list of str
+        Table content extracted from the markdown file.
+    offset : list of int
+        Offset that is used to adjust the correction of number of dashes in the first (offset[0]) and 
+        second (offset[1]) columns by the number specified in percentage. Defaults to [0.0, 0.0].
+    debug : bool
+        If True, print debugging information. Defaults to False.
+
+    Returns
+    -------
+    new_table : list of list of str
+        List of corrected lines of the input table with corrected number of dashes and aligned fences.
+        To be later joined with pipe characters (``|``).
+    """
+    nb_of_rows = len(table)
+    nb_of_cols = len(table[0]) - 2
+
+    nb_of_chars = []
+    for i, row in enumerate(table):
+         # Ignore number of dashes in the count of characters
+        if i != 1:
+            nb_of_chars.append([len(elem) for elem in row])
+
+    # Convert the list to a numpy array and computes the maximum number of chars for each column
+    nb_of_chars_arr = np.array(nb_of_chars)
+    max_chars_in_cols = nb_of_chars_arr.max(axis=0)
+    max_chars = max_chars_in_cols.max()
+
+    # Computes an equal number of dashes per column based on the maximal number of characters over the columns
+    nb_of_dashes = max_chars
+    prop_of_dashes = 1.0 / nb_of_cols
+
+    # Adjust number of characters in first and second column based  offset parameter
+    first_column_width = int(offset[0] * nb_of_dashes) + nb_of_dashes
+    second_column_width = int(offset[1] * nb_of_dashes) + nb_of_dashes

```

Comment:
I simplified the correction of the number dashes to be initially equaled to the maximal number of characters over all the columns, such that we share evenly the page width to each of column. 
In a second step an offset can still be applied (now in percentage) to adjust the first and second columns by a certain percentage. Note that here I use the default value (``offset=[0.0,0.0``]) i.e. no column adjustment.
This also allows the removal of these lines:
```
# Number of chars maximal in one line approximated from a line of the PDF
NB_CHARS_LINE_PDF = 100
``` 
that resolves https://github.com/bids-standard/bids-specification/pull/560#discussion_r465831813